crates/ruff_python_parser/src/string.rs

```diff
@@ -523,20 +498,27 @@ pub enum FStringErrorType {
     UnterminatedString,
     /// Unterminated triple-quoted string.
     UnterminatedTripleQuotedString,
+    // TODO(dhruvmanila): The parser can't catch all cases of this error, but
+    // wherever it can, we'll display the correct error message.
+    /// A lambda expression without parentheses was encountered.
+    LambdaWithoutParentheses,
 }
 
 impl std::fmt::Display for FStringErrorType {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
         use FStringErrorType::{
-            InvalidConversionFlag, SingleRbrace, UnclosedLbrace, UnterminatedString,
-            UnterminatedTripleQuotedString,
+            InvalidConversionFlag, LambdaWithoutParentheses, SingleRbrace, UnclosedLbrace,
+            UnterminatedString, UnterminatedTripleQuotedString,
         };
         match self {
             UnclosedLbrace => write!(f, "expecting '}}'"),
             InvalidConversionFlag => write!(f, "invalid conversion character"),
             SingleRbrace => write!(f, "single '}}' is not allowed"),
             UnterminatedString => write!(f, "unterminated string"),
             UnterminatedTripleQuotedString => write!(f, "unterminated triple-quoted string"),
+            LambdaWithoutParentheses => {
+                write!(f, "lambda expressions are not allowed without parentheses")
```

Comment:
```suggestion
                write!(f, "lambda expressions are not allowed in f-strings without parentheses")
```

Reply:
All variants of `FStringErrorType` gets prefixed with "f-string: ", so the message would be:
```
f-string: lambda expressions are not allowed without parentheses
```